### ISSUE TITLE
require.resolve usage?

### ISSUE THREAD

**Issue by wasd171** [jonaskello/tsconfig-paths#8]
Hello!

Sometimes it's quite handy to use require.resolve for getting absolute paths of some non-script files (databases, images, etc). Unfortunately it seems that `tsconfig-paths` does not support such resolving. Do you have any plans for adding it though?

**Comment by jonaskello** [comment]
The primary scope of this project is to resolve what `tsc` resolves. Could you give an example of how you would like to use it? AFAIK resolving images etc. could only be done if you are using webpack with a loader or something like that?

**Comment by lddubeau** [comment]
@jonaskello "What `tsc` resolves" is an open set, really. For instance, I do here and there `import * as someSchema from "app/some-schema.json"` which `tsc` is completely happy with, so long as there is something that defines a TypeScript type for the module `app/some-schema.json`. The module loader or bundler just needs to be configured so as to produce a result concordant with the module declared in TypeScript when this module is loaded. While `.json` might possibly be the most common case, the same *could* be done with any other extension.

It so happens that I use your package to allow my Mocha tests to find the modules it needs to find, and Node supports loading `.json` files directly. If in the test suite I try to load a `.json` file which is part of the tested application with the currently released version of your package (2.2.0), it fails. If I try with the code on `master`, it works. Node does not support doing a `require` call on `.gif` files, but it would be *possible* to add a package that allows Node to load `.gif` files as modules.

----

I actually came across this issue while running into a problem related to this one. I have an application that is pre-compiled to AMD modules before Mocha runs. I get Node to load the AMD modules by using the `amd-loader` package. However, I discovered that with 2.2.0 the transformations done by `tsconfig-paths` were invisible to `amd-loader`. Then when I tried with the code on `master`, it worked fine.

"Problem solved", as far as I can see it.
